alchemist/alchemist-cognitive-agents/src/main/kotlin/it/unibo/alchemist/model/cognitiveagents/characteristics/cognitive/AbstractCognitiveCharacteristic.kt

```diff
@@ -1,18 +1,24 @@
 package it.unibo.alchemist.model.cognitiveagents.characteristics.cognitive
 
-import com.google.common.primitives.Doubles
 import com.uchuhimo.konf.Config
 import it.unibo.alchemist.model.cognitiveagents.characteristics.PARAMETERS_FILE
 
+/**
+ * The generic implementation of a cognitive characteristic.
+ */
 abstract class AbstractCognitiveCharacteristic : CognitiveCharacteristic {
 
+    /**
+     * The current level of this characteristic.
+     */
     protected var currentLevel: Double = 0.0
-        set(value) {
-            field = Doubles.constrainToRange(value, 0.0, 1.0)
-        }
 
     override fun level() = currentLevel
 
+    /**
+     * The function according to, the parameters involved
+     * in this characteristic update are combined.
```

Comment:
This is unclear to me

Reply:
It's the algorithm which decides how the parameters relevant for this characteristic must be combined together. It can be either a max, min, mean or whatever else.

Reply:
I mean: the description is not clear. Expand it and make it clearer

Reply:
fixed